apparmor.d/profiles-a-f/briar-desktop

```diff
@@ -0,0 +1,97 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2024 Roman Beslik <me@beroal.in.ua>
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = @{bin}/briar-desktop
+profile briar-desktop @{exec_path} {
+  include <abstractions/audio-client>
+  include <abstractions/base>
+  include <abstractions/dconf-write>
+  include <abstractions/fontconfig-cache-read>
+  include <abstractions/gnome-strict>
+  include <abstractions/graphics>
+  include <abstractions/nameservice-strict>
+
+  network inet dgram,
+  network inet stream,
+  network inet6 dgram,
+  network inet6 stream,
+
+  @{lib}/jvm/java*/lib/jspawnhelper Cx -> jspawnhelper,
+
+  ptrace read peer=briar-desktop-tor,
+  ptrace read peer=@{profile_name}//jspawnhelper,
+
+  @{exec_path} mr,
+
+  @{lib}/jvm/java*/bin/java rix,
+  @{lib}/jvm/java*/lib/** rm,
+  @{sh_path} mr,
+
+  @{system_share_dirs}/icons/{,**} r,
+  @{system_share_dirs}/java/briar-desktop.jar r,
+
+  /etc/java*/{{logging,net,sound}.properties,security/java.{security,policy}} r,
+
+  owner @{HOME}/.briar/{,**} rw,
+  owner @{HOME}/.briar/desktop/db/db.mv.db k,
+
+  owner @{HOME}/.java/{,.userPrefs/{,org/}} w,
+  owner @{HOME}/.java/.userPrefs/.user.lock.@{user} wk,
+  owner @{HOME}/.java/.userPrefs/.userRootModFile.@{user} rw,
+  owner @{HOME}/.java/.userPrefs/{,org/}prefs.{xml,tmp} rw,
+  owner @{HOME}/.java/.userPrefs/org/briarproject/{,**} rw,
+
+  owner @{HOME}/.skiko/ w,
+  owner @{HOME}/.skiko/@{hex64}/{,libskiko-*.so,skiko[0-9]*} mrw,
+
+  owner @{user_pictures_dirs}/{,**} r,
+
+  owner @{user_cache_dirs}/JNA/{,**} mrw,
+
+  owner @{tmp}/hsperfdata_@{user}/ rw,
+  owner @{tmp}/hsperfdata_@{user}/@{pid} rwk,
+  owner @{tmp}/imageio[0-9]*.tmp rw,
+  owner @{tmp}/jna[0-9]*.tmp mrw,
```

Comment:
Use `@{int}` instead of `[0-9]*`

Reply:
I tried. The number is too long.

Reply:
How long? https://apparmor.pujol.io/variables/#basic-variables-of-a-given-length

Reply:
19-20 digits. I try to match the numbers with `@{int64}`, but it doesn't work for an unknown reason. The log:

```
type=AVC msg=audit(1728039823.562:27797): apparmor="DENIED" operation="unlink" class="file" profile="briar-desktop" name="/tmp/imageio18417488744924248646.tmp" pid=44544 comm="Finalizer" requested_mask="d" denied_mask="d" fsuid=1020 ouid=1020FSUID="test58" OUID="test58"
type=AVC msg=audit(1728039823.562:27798): apparmor="DENIED" operation="unlink" class="file" profile="briar-desktop" name="/tmp/imageio16063985725205008127.tmp" pid=44544 comm="Finalizer" requested_mask="d" denied_mask="d" fsuid=1020 ouid=1020FSUID="test58" OUID="test58"
type=AVC msg=audit(1728039823.562:27799): apparmor="DENIED" operation="unlink" class="file" profile="briar-desktop" name="/tmp/imageio3075731188056589370.tmp" pid=44544 comm="Finalizer" requested_mask="d" denied_mask="d" fsuid=1020 ouid=1020FSUID="test58" OUID="test58"
type=AVC msg=audit(1728039873.916:27806): apparmor="DENIED" operation="mknod" class="file" profile="briar-desktop" name="/tmp/imageio6276895853312334707.tmp" pid=44544 comm="Thread-13" requested_mask="c" denied_mask="c" fsuid=1020 ouid=1020FSUID="test58" OUID="test58"
type=AVC msg=audit(1728039880.416:27807): apparmor="DENIED" operation="mknod" class="file" profile="briar-desktop" name="/tmp/imageio15645067652751410968.tmp" pid=44544 comm="Thread-14" requested_mask="c" denied_mask="c" fsuid=1020 ouid=1020FSUID="test58" OUID="test58"
type=AVC msg=audit(1728039886.966:27808): apparmor="DENIED" operation="mknod" class="file" profile="briar-desktop" name="/tmp/imageio5235771028765679424.tmp" pid=44544 comm="Thread-15" requested_mask="c" denied_mask="c" fsuid=1020 ouid=1020FSUID="test58" OUID="test58"
type=AVC msg=audit(1728039891.983:27809): apparmor="DENIED" operation="mknod" class="file" profile="briar-desktop" name="/tmp/imageio4098760449481265574.tmp" pid=44544 comm="Thread-16" requested_mask="c" denied_mask="c" fsuid=1020 ouid=1020FSUID="test58" OUID="test58"
type=AVC msg=audit(1728039945.737:27818): apparmor="DENIED" operation="mknod" class="file" profile="briar-desktop" name="/tmp/imageio6588868399424867249.tmp" pid=44544 comm="Thread-17" requested_mask="c" denied_mask="c" fsuid=1020 ouid=1020FSUID="test58" OUID="test58"
```

Reply:
I see that `@{int64}` means “64 decimal digits”, but I need “a 64-bit unsigned number in the decimal notation”.

Reply:
Yes, that would be `@{u64}`, witch does not exist yet. There is `@{u8}` & `@{u16}`, I need to make the other. 
Simply put `@{u64}`. the CI will fail, but I will add it. However, in apparmor notation, that is the same than `@{int}@{int}`

Reply:
Not quite. `@{int}@{int}` matches at least two digits.

Reply:
True, but when you need more than 10 digits. It is usually not an issue (the definition of `@{u*}` are painful, and not as strict as they should be. See [18a7151](https://github.com/roddhjav/apparmor.d/commit/18a71512a9519585216f6fec11b4b7371630990f)